tests/expr_and_series/str/zfill_test.py

```diff
@@ -0,0 +1,39 @@
+from __future__ import annotations
+
+import pytest
+
+import narwhals as nw
+from tests.utils import Constructor, ConstructorEager, assert_equal_data
+
+data = {"a": ["-1", "+1", "1", "12", "123", "99999", "+9999", None]}
+expected = {"a": ["-01", "+01", "001", "012", "123", "99999", "+9999", None]}
+
+
+def skip_pandas_pyarrow(constructor: Constructor | ConstructorEager) -> None:
+    name: str = constructor.__name__
+    if name in {"pandas_pyarrow_constructor", "modin_pyarrow_constructor"}:
+        reason = (
+            "pandas with pyarrow backend doesn't support str.zfill, see "
+            "https://github.com/pandas-dev/pandas/issues/61485"
+        )
+        raise pytest.skip(reason=reason)
+
+    if "polars" in name:
+        reason = "Polars zfill behavior is different from pandas at the moment."
+        raise pytest.skip(reason=reason)
```

Comment:
can we in-line this (like we do for other tests)?

and also, for Polars, we should test it. in `PolarsExprStringNamespace`, we could implement this for Polars with version checks, kind of like we do for `PolarsExprListNamespace.len`, where for old Polars you replicate the logic with `pad` / `when/then`, and for newer versions you just use `str.zfill`

Reply:
What do you mean by in-line? Can you share an example